pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -507,12 +516,98 @@ func updateCertTimer(configInterval uint32, tickerHandle interface{}) {
 	flextimer.TickNow(tickerHandle)
 }
 
+func decryptAuthContainer(getconfigCtx *getconfigContext, sendRV *zedcloud.SendRetval) error {
+	decryptCtx := &cipher.DecryptCipherContext{
+		Log:                  log,
+		AgentName:            agentName,
+		PubSubControllerCert: getconfigCtx.pubControllerCert,
+		PubSubEdgeNodeCert:   getconfigCtx.zedagentCtx.subEdgeNodeCert,
+	}
+	sm := &zauth.AuthContainer{}
+	err := proto.Unmarshal(sendRV.RespContents, sm)
+	if err != nil {
+		return err
+	}
+	cipherBlock := sm.GetCipherData()
+	if cipherBlock == nil {
+		// Nothing encrypted, nothing to do
+		return nil
+	}
+	// Verify auth container and propagate the certificate
+	// errors to the caller by updating the retval status.
+	// That is utterly important that a caller schedules
+	// certs update in that case.
+	status, err := zedcloud.VerifyAuthContainer(zedcloudCtx, sm)
+	if err != nil {
+		sendRV.Status = status
+		return err
+	}
+	cipherContext := sm.GetCipherContext()
+	if decryptCtx == nil {
```

Comment:
How can it become `nil`? It was just allocated on line 520, wasn't it?

Reply:
Left-over from the previous version of the function, which accepted that as the param. Will remove, thanks.